### ISSUE TITLE
VSTest adapter delays solution load

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#4368]
```
0.74%   RebuildTests  •  655 ms  •  1 call  •  MonoDevelop.UnitTesting.UnitTestService.RebuildTests
  0.74%   BuildTest  •  655 ms  •  1 call  •  MonoDevelop.UnitTesting.UnitTestService.BuildTest(WorkspaceObject)
    0.74%   CreateUnitTest  •  654 ms  •  1 call  •  MonoDevelop.UnitTesting.SystemTestProvider.CreateUnitTest(WorkspaceObject)
      0.73%   get_HasTests  •  649 ms  •  1 call  •  MonoDevelop.UnitTesting.UnitTestGroup.get_HasTests
        0.73%   get_Tests  •  649 ms  •  1 call  •  MonoDevelop.UnitTesting.UnitTestGroup.get_Tests
          0.73%   OnCreateTests  •  649 ms  •  1 call  •  MonoDevelop.UnitTesting.SolutionFolderTestGroup.OnCreateTests
            0.73%   BuildTest  •  649 ms  •  6 calls  •  MonoDevelop.UnitTesting.UnitTestService.BuildTest(WorkspaceObject)
              0.73%   CreateUnitTest  •  649 ms  •  6 calls  •  MonoDevelop.UnitTesting.SystemTestProvider.CreateUnitTest(WorkspaceObject)
                0.73%   get_HasTests  •  645 ms  •  5 calls  •  MonoDevelop.UnitTesting.UnitTestGroup.get_HasTests
                  0.73%   get_Tests  •  645 ms  •  5 calls  •  MonoDevelop.UnitTesting.UnitTestGroup.get_Tests
                    0.73%   OnCreateTests  •  645 ms  •  5 calls  •  MonoDevelop.UnitTesting.SolutionFolderTestGroup.OnCreateTests
                      0.73%   BuildTest  •  645 ms  •  63 calls  •  MonoDevelop.UnitTesting.UnitTestService.BuildTest(WorkspaceObject)
                        0.50%   CreateUnitTest  •  439 ms  •  46 calls  •  MonoDevelop.UnitTesting.VsTest.VsTestTestProvider.CreateUnitTest(WorkspaceObject)
                          0.49%   GetTestAdapters  •  439 ms  •  28 calls  •  MonoDevelop.UnitTesting.VsTest.VsTestAdapter.GetTestAdapters(Project)
                             0.44%   GetFiles  •  389 ms  •  270 calls  •  System.IO.Directory.GetFiles(String, String, SearchOption)
                             0.05%   16 functions hidden  •  49 ms total  •  877 calls total
                           0.00%   3 functions hidden  •  0 ms total  •  30 calls total
                        0.21%   CreateUnitTest  •  188 ms  •  63 calls  •  MonoDevelop.UnitTesting.SystemTestProvider.CreateUnitTest(WorkspaceObject)
                          0.20%   get_HasTests  •  176 ms  •  26 calls  •  MonoDevelop.UnitTesting.UnitTestGroup.get_HasTests
                            0.20%   get_Tests  •  176 ms  •  26 calls  •  MonoDevelop.UnitTesting.UnitTestGroup.get_Tests
                              0.20%   OnCreateTests  •  176 ms  •  26 calls  •  MonoDevelop.UnitTesting.SolutionFolderTestGroup.OnCreateTests
                                0.20%   BuildTest  •  175 ms  •  78 calls  •  MonoDevelop.UnitTesting.UnitTestService.BuildTest(WorkspaceObject)
                                  0.18%   CreateUnitTest  •  156 ms  •  68 calls  •  MonoDevelop.UnitTesting.VsTest.VsTestTestProvider.CreateUnitTest(WorkspaceObject)
                                    0.18%   GetTestAdapters  •  156 ms  •  67 calls  •  MonoDevelop.UnitTesting.VsTest.VsTestAdapter.GetTestAdapters(Project)
                                       0.11%   GetFiles  •  100 ms  •  114 calls  •  System.IO.Directory.GetFiles(String, String, SearchOption)
                                       0.06%   16 functions hidden  •  56 ms total  •  1032 calls total
                                     0.00%   IsNullOrEmpty  •  0 ms  •  67 calls  •  System.String.IsNullOrEmpty(String)
                                   0.02%   5 functions hidden  •  19 ms total  •  750 calls total
                                 0.00%   7 functions hidden  •  0 ms total  •  280 calls total
                              ►0.00%   UnitTestCollection..ctor  •  0 ms  •  26 calls  •  MonoDevelop.UnitTesting.UnitTestCollection..ctor(UnitTestGroup)
                             0.00%   5 functions hidden  •  0 ms total  •  94 calls total
                           0.01%   2 functions hidden  •  12 ms total  •  44 calls total
                         0.02%   4 functions hidden  •  17 ms total  •  492 calls total
                       0.00%   7 functions hidden  •  0 ms total  •  180 calls total
                    ►0.00%   UnitTestCollection..ctor  •  0 ms  •  5 calls  •  MonoDevelop.UnitTesting.UnitTestCollection..ctor(UnitTestGroup)
                   0.00%   6 functions hidden  •  0 ms total  •  19 calls total
                 0.00%   2 functions hidden  •  4 ms total  •  8 calls total
               0.00%   5 functions hidden  •  0 ms total  •  46 calls total
             0.00%   7 functions hidden  •  0 ms total  •  20 calls total
          ►0.00%   UnitTestCollection..ctor  •  0 ms  •  1 call  •  MonoDevelop.UnitTesting.UnitTestCollection..ctor(UnitTestGroup)
         0.00%   5 functions hidden  •  0 ms total  •  5 calls total
       0.01%   2 functions hidden  •  5 ms total  •  2 calls total
     0.00%   3 functions hidden  •  0 ms total  •  3 calls total
   0.00%   10 functions hidden  •  0 ms total  •  11 calls total
```

> VS bug [#591905](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/591905)

**Comment by Therzok** [comment]
On the other hand, there's also the possibility to create the unit tests in a background thread, as the project model is safe for reading.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-52] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4795 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 5a6b7c63212b801c6a1e80e68d5403d5abe49373 (xamarin/monodevelop)
